src/price/crypto-chart-box.tsx

```diff
@@ -0,0 +1,472 @@
+    cursor: 'pointer',
+    display: 'flex',
+    background: 'none',
+    border: 'none',
+    outline: 'none',
+    float: 'right'
+  },
+
+  buttonViewMore: {
+    outline: 'none',
+    fontSize: '12px',
+    color: '#93b0c1',
+    cursor: 'pointer',
+    textTransform: 'uppercase'
+  },
+
+  buttonViewMoreText: {
+    borderBottom: '1px dashed #93b0c1'
+  },
+
+  expandMore: {
+    verticalAlign: 'middle !important'
+  },
+
+  header: {
+    margin: '10px'
+  },
+
+  headerText: {
+    fontSize: '20px',
+    lineHeight: 1.8,
+    opacity: 0.83
+  }
+};
+
+export type Token = {
+  name: string,
+  symbol: string,
+  balance: number,
+  balanceInFiat: number
+}
+
+export type CryptoChartBoxProps = {
+  locale: string,
+  fiatCurrency: string,
+  tokens: Array<Token>,
+  manageCryptoAction?: ((event: React.MouseEvent<HTMLElement>) => void) 
+}
+
+export type StyledProps = WithStyles<keyof typeof styles> & CryptoChartBoxProps;
+
+export type Active = {
+  active: boolean;
+};
+
+export type CryptoChartBoxState = {
+  activations: Array<Active>,
+  displayedTokens: Array<Token>,
+  viewAll: boolean
+};
+
+export type ChartType = {
+  setSelection: Function
+}
+
+export type ChartElementType = {
+  wrapper: ChartWrapperType
+}
+
+export type ChartWrapperType = {
+  getChart: Function
+}
+
+export class CryptoChartBoxComponent extends React.Component<StyledProps, CryptoChartBoxState> {
+  
+  TOP_TOKEN_LIST_SIZE = 5;
+  OTHERS_COLOR = '#71a6b8';
+  otherTokens: Array<Token> = [];
+
+  activations: Array<Active> = [];
+  topTokens: Array<Token> = [];
+  othersToken: Token = {
+    name: 'Others',
+    symbol: 'OTHERS',
+    balance: 0,
+    balanceInFiat: 0
+  };
+
+  state = {
+    activations: this.activations,
+    displayedTokens: this.topTokens,
+    viewAll: true
+  }
+
+  selection = [];
+  chart: ChartType;
+
+  public refs: {
+    pieChart: HTMLElement & ChartElementType;
+  };
+  
+  constructor(props: StyledProps) {
+    super(props);
+
+    this.initState(props);
+    this.initSelection();
+  }
+
+  initState(props: StyledProps) {
+    if (!props.tokens) {
+      return;
+    }
+    props.tokens.forEach((token, index) => {
+      this.state.activations[index] = {active: false};
+    });
+
+    this.topTokens = props.tokens.slice(0, this.TOP_TOKEN_LIST_SIZE);
+    this.otherTokens = props.tokens.slice(this.TOP_TOKEN_LIST_SIZE, props.tokens.length);
+    if (this.otherTokens.length) {
+      this.othersToken = this.getOthersToken(this.otherTokens);
+      this.state.displayedTokens = [...this.topTokens, this.othersToken]; 
+    } else {
+      this.state.displayedTokens = this.topTokens;
+    }
+  }
+
+  getOthersToken(otherTokens: Array<Token>){
+    return {
+      name: 'Others',
+      symbol: 'OTHERS',
+      balance: this.getOthersTokenBalance(otherTokens, 'balance'),
+      balanceInFiat: this.getOthersTokenBalance(otherTokens, 'balanceInFiat')
+    };
+  }
+
+  getOthersTokenBalance(otherTokens: Array<Token>, balanceType: string) {
```

Comment:
These functions should be in the reducer and wrapped in a [memoized selector](https://redux.js.org/recipes/computing-derived-data) for performance reasons. If they're implemented in the component like this the values will be recomputed every time the component is mounted or unmounted.

Reply:
@dfcowell I thought about that and I was thinking to do that when we will implement the Redux for price update. When thinking again about this I see that there is a way to integrate it with Redux right now. I can move this to the reducer and call the action on a didMount on the Wrapper Component in the Wallet. So when angular changes the data and the props updating the wrapper component it will call the action which will run the reducer. Let me know if that does not make sense.

Reply:
I keep forgetting that we're not integrating Redux yet. It looks like doing this one component at a time is creating a lot of re-work as we will have to go back and refactor a bunch of this new code.

Would it be worth integrating redux *now* rather than continuing on with the current approach?